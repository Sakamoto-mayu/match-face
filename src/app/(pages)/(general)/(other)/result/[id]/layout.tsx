export const metadata = {
  title: "回答確認",
  description: "一般ユーザー回答確認ページ",
};

export default function ResultLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
