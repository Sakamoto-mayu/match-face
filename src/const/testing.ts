export const project = [
  {
    id: "1",
    name: "案件のタイトルが入ります",
    date: new Date("2023/07/01"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "2",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-02"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "3",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-03"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
];

export const question = [
  {
    id: "1",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "2",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "3",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "4",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "5",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "6",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "7",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "8",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "9",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
];
