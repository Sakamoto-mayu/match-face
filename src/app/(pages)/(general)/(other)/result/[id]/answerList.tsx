import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "@/components/pages/general/result/ModelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}:{question.name}
          </h3>
          <p id={`content-${index + 1}`} className="mt-2 mb-4">
            {question.content}
          </p>
          {question.type === "writing" ? (
            <div
              id={`answer-${index + 1}`}
              className="border border-gray p-3 mt-2 rounded-md shadow-md"
            >
              <h3 className="mb-2 text-xl">あなたの回答</h3>
              {currentAnswer[index].content}
            </div>
          ) : (
            <>
              <ul>
                {question.choices?.map((choice, index) => (
                  <li key={index}>・{choice}</li>
                ))}
              </ul>
              <div
                id={`answer-${index + 1}`}
                className="border border-gray p-3 mt-5 rounded-md shadow-md"
              >
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
