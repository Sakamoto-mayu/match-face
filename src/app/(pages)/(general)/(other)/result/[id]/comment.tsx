import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import { User } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: number;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );
  const commentUser = User.filter(
    (user) => user.id === currentComment[0].admin_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">{commentUser[0].name}からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={commentUser[0].image_url}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
            priority
          />
        </div>
        <div id="comment" className="w-5/6">
          {currentComment[0].content}
        </div>
      </div>
    </>
  );
};
