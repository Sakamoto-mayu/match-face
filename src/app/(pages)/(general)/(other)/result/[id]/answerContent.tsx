"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";
import { Answer } from "@/const/result";
import { useRouter } from "next/navigation";

type Props = {
  user_id: number;
  project_id: number;
};

export const AnswerContent = ({ user_id, project_id }: Props) => {
  const router = useRouter();
  const currentProject = project.filter(
    (project) => project.id === Number(project_id)
  );
  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === Number(project_id)
  );

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        {/*現状のデータでは4以降の解答データを呼び出そうとするとエラーになります */}
        <p>回答日時:{currentAnswer[0].answer_date.toLocaleDateString()}</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">案件名：{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList project_id={project_id} />
      <CommentContent user_id={user_id} project_id={project_id} />
      <div className="flex justify-around my-20">
        <OrangeButton
          className="border rounded-none"
          label="質問一覧へ"
          onClick={() => router.push("/questions")}
        />
        <OrangeButton
          className="border rounded-none"
          label="回答履歴一覧へ"
          onClick={() => router.push("/histories")}
        />
      </div>
    </div>
  );
};
