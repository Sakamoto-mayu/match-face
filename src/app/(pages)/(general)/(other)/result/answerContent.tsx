"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";

export const AnswerContent = ({ id }: { id: number }) => {
  const currentProject = project.filter((project) => project.id === id);

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        <p>回答日時：</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList id={id} />
      <CommentContent />
      <div className="flex justify-around my-20">
        <OrangeButton label="質問一覧へ" />
        <OrangeButton label="回答履歴一覧へ" />
      </div>
    </div>
  );
};
