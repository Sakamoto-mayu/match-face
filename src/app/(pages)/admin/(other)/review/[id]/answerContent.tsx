"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";
import GrayButton from "@/components/ui/button/GrayButton";
import { Answer } from "@/const/review";
import { useRouter } from "next/navigation";
import ConfirmationModal from "@/components/ui/ConfirmationModal";
import { useState } from "react";

type Props = {
  user_id: string;
  project_id: string;
  admin_id: string;
};

export const AnswerContent = ({ user_id, project_id, admin_id }: Props) => {
  const router = useRouter();
  const [isOpen, setIsOpen] = useState(false);
  const currentProject = project.filter((project) => project.id === project_id);
  const currentAnswer = Answer.filter((answer) => project_id == project_id);

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        <p>回答日時：{currentAnswer[0].answer_date.toLocaleDateString()}</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">案件名：{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList user_id={user_id} project_id={project_id} />
      <CommentContent
        user_id={user_id}
        project_id={project_id}
        admin_id={admin_id}
      />
      <div className="flex justify-around my-20">
        <ConfirmationModal
          isOpen={isOpen}
          setIsOpen={setIsOpen}
          message="コメントを送信してよろしいですか？"
          firstLabel="コメントを送信する"
          secondLabel="キャンセル"
        />
        <OrangeButton
          label="コメントを送信する"
          className="w-50 border rounded-none"
          onClick={() => setIsOpen(true)}
        />
        <GrayButton
          className="border rounded-none w-50"
          label="回答履歴一覧へ戻る"
          onClick={() => router.push("/admin/histories")}
        />
      </div>
    </div>
  );
};
