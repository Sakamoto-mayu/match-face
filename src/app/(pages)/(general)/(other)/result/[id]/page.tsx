import { AnswerContent } from "./answerContent";

const ResultPage = ({ params }: { params: { id: number } }) => {
  const user_id = 1;

  return (
    <div className="flex flex-col items-center">
      <AnswerContent user_id={user_id} project_id={params.id} />
    </div>
  );
};

export default ResultPage;
