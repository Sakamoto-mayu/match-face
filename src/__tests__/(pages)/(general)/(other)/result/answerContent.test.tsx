import { render, screen } from "@testing-library/react";
import { AnswerContent } from "@/app/(pages)/(general)/(other)/result/[id]/answerContent";
import { AppRouterContextProviderMock } from "../../../../test_utils/app-router-context-provider-mock";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { project } from "@/const/testing";

const user = userEvent.setup();

describe("AnswerContentコンポーネントのテスト", () => {
  const testData = {
    user_id: "user1",
    project_id: "1",
    answerDate: "2023/7/6",
    projectTitle: project[0].name,
    projectContent: project[0].project_detail,
  };

  test("回答日時が表示される", () => {
    const push = jest.fn();
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <AnswerContent
          user_id={testData.user_id}
          project_id={testData.project_id}
        />
      </AppRouterContextProviderMock>
    );
    const answerDate = screen.getByText(`回答日時：${testData.answerDate}`);
    expect(answerDate).toBeInTheDocument();
  });

  test("案件名・案件概要が表示される", () => {
    const push = jest.fn();
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <AnswerContent
          user_id={testData.user_id}
          project_id={testData.project_id}
        />
      </AppRouterContextProviderMock>
    );
    const projectTitle = screen.getByRole("heading", { level: 2 });
    expect(projectTitle).toHaveTextContent(`案件名：${testData.projectTitle}`);

    const projectContent = screen.getByText(testData.projectContent);
    expect(projectContent).toBeInTheDocument();
  });

  test("質問一覧へのボタンが正常に動く", async () => {
    const push = jest.fn();
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <AnswerContent
          user_id={testData.user_id}
          project_id={testData.project_id}
        />
      </AppRouterContextProviderMock>
    );
    const button = screen.getByRole("button", { name: "質問一覧へ" });
    expect(button).toBeInTheDocument();

    await user.click(button);

    expect(push).toHaveBeenCalledWith("/questions");
  });

  test("回答履歴一覧へのボタンが正常に動く", async () => {
    const push = jest.fn();
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <AnswerContent
          user_id={testData.user_id}
          project_id={testData.project_id}
        />
      </AppRouterContextProviderMock>
    );
    const button = screen.getByRole("button", { name: "回答履歴一覧へ" });
    expect(button).toBeInTheDocument();

    await user.click(button);
    expect(push).toHaveBeenCalledWith("/histories");
  });

  test("スナップショットテスト", () => {
    const push = jest.fn();

    const contents = render(
      <AppRouterContextProviderMock router={{ push }}>
        <AnswerContent
          user_id={testData.user_id}
          project_id={testData.project_id}
        />
      </AppRouterContextProviderMock>
    );
    expect(contents).toMatchSnapshot();
  });
});
