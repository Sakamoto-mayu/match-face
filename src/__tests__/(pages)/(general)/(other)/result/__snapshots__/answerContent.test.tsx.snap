// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショットテスト 1`] = `
<div>
  <div
    class="w-1/2 py-5"
  >
    <div
      class="flex justify-between items-center my-3"
    >
      <h1
        class="text-2xl"
      >
        回答の確認
      </h1>
      <p>
        回答日時:
        2023/7/6
      </p>
    </div>
    <hr
      class="border border-black"
    />
    <h2
      class="text-xl my-2"
    >
      案件名：
      案件のタイトルが入ります
    </h2>
    <p
      class="text-lg"
    >
      案件の詳細内容が入ります。テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
    </p>
    <div
      class="py-5"
    >
      <h3
        class="text-xl"
      >
        Q
        1
        :
        質問のタイトルが入ります
      </h3>
      <p
        class="mt-2 mb-4"
        id="content-1"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <div
        class="border border-gray p-3 mt-2 rounded-md shadow-md"
        id="answer-1"
      >
        <h3
          class="mb-2 text-xl"
        >
          あなたの回答
        </h3>
        回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。
      </div>
      <div
        class="my-7 p-3 rounded-md bg-gray-200"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
    <div
      class="py-5"
    >
      <h3
        class="text-xl"
      >
        Q
        2
        :
        質問のタイトルが入ります
      </h3>
      <p
        class="mt-2 mb-4"
        id="content-2"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <ul>
        <li>
          ・
          選択肢A
        </li>
        <li>
          ・
          選択肢B
        </li>
        <li>
          ・
          選択肢C
        </li>
        <li>
          ・
          選択肢D
        </li>
      </ul>
      <div
        class="border border-gray p-3 mt-5 rounded-md shadow-md"
        id="answer-2"
      >
        <h3
          class="mb-2 text-xl"
        >
          あなたの回答
        </h3>
        選択肢A
      </div>
      <div
        class="my-7 p-3 rounded-md bg-gray-200"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
    <div
      class="py-5"
    >
      <h3
        class="text-xl"
      >
        Q
        3
        :
        質問のタイトルが入ります
      </h3>
      <p
        class="mt-2 mb-4"
        id="content-3"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <div
        class="border border-gray p-3 mt-2 rounded-md shadow-md"
        id="answer-3"
      >
        <h3
          class="mb-2 text-xl"
        >
          あなたの回答
        </h3>
        回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。
      </div>
      <div
        class="my-7 p-3 rounded-md bg-gray-200"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
    <h3
      class="text-xl mb-3"
    >
      管理者太郎
      からのコメント
    </h3>
    <div
      class="border border-black rounded-md p-3 flex justify-between "
    >
      <div
        class="w-1/6 flex justify-center items-center"
      >
        <img
          alt="コメントしたユーザーのアイコン"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="70"
          src="/_next/image?url=%2Ficon%2Fhuman_icon.png&w=256&q=75"
          srcset="/_next/image?url=%2Ficon%2Fhuman_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fhuman_icon.png&w=256&q=75 2x"
          style="color: transparent;"
          width="70"
        />
      </div>
      <div
        class="w-5/6"
        id="comment"
      >
        コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。
      </div>
    </div>
    <div
      class="flex justify-around my-20"
    >
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 w-40 h-auto text-lg border rounded-none"
        >
          質問一覧へ
        </button>
      </div>
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 w-40 h-auto text-lg border rounded-none"
        >
          回答履歴一覧へ
        </button>
      </div>
    </div>
  </div>
</div>
`;
